### PR TITLE
refactor: make `nestedTypesOf` tail recursive

### PR DESCRIPTION
**Pull request by JonathanStarup**
Should be faster, because of the removed flatMap

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Reducer.scala [hunk 1]
```diff
@@ -265,52 +267,41 @@ object Reducer {
         // the enum type itself
         val eType = e.tpe
         // the types inside the cases
-        val caseTypes = e.cases.values.flatMap(c => nestedTypesOf(c.tpe)(root, flix))
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Reducer.scala:268]: This is double work so I removed the call

**Comment by magnus-madsen** [comment]
But the real solution, of course, is to remove the visit entirely right? Instead we should do it as part of the normal traversal.

**Comment by magnus-madsen** [comment]
Any impact on performance with `Xperf --n 17`?

**Comment by JonathanStarup** [comment]
PR Reducer: 465.9578
Master Reducer: 564.29455

**Comment by magnus-madsen** [comment]
!!!jar

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**